include/odp/api/spec/schedule.h

```diff
@@ -45,6 +46,53 @@ extern "C" {
  * Maximum schedule group name length in chars including null char
  */
 
+/**
+ * Schedule configuration options
+ */
+typedef struct odp_schedule_config_t {
+
+	/* Scheduler flow aware
+	 * Application can configure the scheduler to be flow aware or unaware.
+	 * When the scheduler is configured as flow aware the ORDERED and
+	 * ATOMIC schedule queue synchronization by the scheduler is performed
+	 * per flow within each queue.
+	 */
+	odp_support_t flow_aware;
+
+	/* Number of parallel flows per queue to be supported
+	 * Only valid when scheduler is configured as flow_aware
+	 *
+	 * Flows are lightweight entity and packets can be assigned to specific
+	 * flow by the application using odp_packet_flow_hash_set() before
+	 * enqueuing the packet into the scheduler.
+	 * This value is valid only when scheduler is configured in flow aware
+	 * mode.
+	 * Depening on the implementation this number might be rounded-off to
+	 * nearest supported value (e.g power of 2)
+	 * This number should be less than maximum flow supported by the
+	 * implementation.
+	 * @see odp_schedule_capability_t
+	 */
+	uint32_t flow_count;
+
+	/* Maximum number of queues to be supported
+	 * Application configures the maximum number of queues to be supported
+	 * by the implementation. This is a combination of all queue types.
+	 * @see odp_queue_capability_t
+	 */
+	uint32_t queue_count;
+} odp_schedule_config_t;
+
+typedef struct odp_schedule_capability_t {
+
+	/* Maximum supported flow
+	 * Specifies the maximum number of flows supported by the
+	 * implementation. A value of 1 indicates the schedule does not
+	 * support flow aware mode.
+	 */
+	uint32_t max_flow;
```

Comment:
what is this number for your hardware? should it be big number?

Reply:
@muvarov since `flow_count` is `uint32_t` it makes sense that `max_flow` would have the same magnitude.

Reply:
Shouldn't `odp_schedule_capability_t` be where whether flow-aware mode is supported as well as the max flows and queues is returned? Basically this struct should return the options and limits that the application then chooses from in its `odp_schedule_config()` call.

Reply:
Do we want to move things like number of scheduler groups here? That would seem to make more sense than burying it in the queue config as @bogdanPricope has previously noted.

Reply:
`odp_schedule_capability_t.max_flow` should probably be per queue to match `odp_schedule_config_t .flow_count` ? If so, it should be clearly stated.